### PR TITLE
Basic implementation of plugin

### PR DESCRIPTION
**Pull request by isnifer**
If you want to test it locally, please set environment variable
```sh
FLAVORS=custom,tipsi
```

Thank you :)

### CODE REVIEW

File: .travis/ci.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+case "${TRAVIS_OS_NAME}" in
+  osx)
+    cd example_tmp
+    set -o pipefail && npm run build:ios | xcpretty -c -f `xcpretty-travis-formatter`
+    npm run test:ios
+    cd ../
+    npm test
```
Comment (itsmepetrov) [inline .travis/ci.sh]: Maybe module tests first and then example tests?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+          - extra-google-google_play_services
+
+cache:
+  directories:
+    - $HOME/.nvm
+    - $HOME/.npm
+    - $HOME/.cocoapods
+    - $HOME/.gradle/caches/
+    - $HOME/.gradle/wrapper/
+    - example/node_modules
```
Comment (itsmepetrov) [inline .travis.yml]: We should cache `example_tmp/node_modules` dir instead

File: .travis/ci.sh (hunk 1 above)
Comment (isnifer) [reply .travis/ci.sh]: Nope, on travis it creates incorrect behavior with environment variables.

File: .travis.yml (hunk 2 above)
Comment (isnifer) [reply .travis.yml]: Yes, sure

**Review by itsmepetrov — APPROVED** [review]
Looks good to merge

**Comment by isnifer** [comment]
@cybergrind Kirill, please look into PR :)

**Comment by cybergrind** [comment]
Got this error
```
:app:bundleReleaseJsAndAssets
[12/27/2016, 6:41:23 PM] <START> Initializing Packager
[12/27/2016, 6:41:23 PM] <START> Building in-memory fs for JavaScript
[12/27/2016, 6:41:23 PM] <END>   Building in-memory fs for JavaScript (115ms)
[12/27/2016, 6:41:23 PM] <START> Building Haste Map
[12/27/2016, 6:41:23 PM] <END>   Building Haste Map (248ms)
[12/27/2016, 6:41:23 PM] <END>   Initializing Packager (415ms)
[12/27/2016, 6:41:23 PM] <START> Transforming files
Warning: The transform cache was reset.

TransformError: /home/kpi/devel/tipsi/babel-plugin-tipsi-flavors/example_tmp/index.android.js: /home/kpi/devel/tipsi/babel-plugin-tipsi-flavors/example_tmp/index.android.js: Maximum call stack size exceeded

:app:bundleReleaseJsAndAssets FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:bundleReleaseJsAndAssets'.
> Process 'command 'node'' finished with non-zero exit value 1

* Try:
Run with --info or --debug option to get more log output.
```

**Comment by isnifer** [comment]
Looks like something wrong with watchman cache.
@cybergrind I talked with @itsmepetrov, he recommended kill watchman cache.
Try:
```bash
watchman watch-del %FULL_PATH_TO_EXAMPLE_TMP_PROJECT%
watchman watch-project %FULL_PATH_TO_EXAMPLE_TMP_PROJECT%
```
And re-run `FLAVORS='custom,tipsi' npm run ci`

**Comment by cybergrind** [comment]
@itsmepetrov looks like that was caused by `npm-proxy-cache`

**Pull request merged**